tests/reg_issue1352/issue_1352.fz

```diff
@@ -0,0 +1,26 @@
+# This file is part of the Fuzion language implementation.
+#
+# The Fuzion language implementation is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as published
+# by the Free Software Foundation, version 3 of the License.
+#
+# The Fuzion language implementation is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion test issue_1352
+#
+# -----------------------------------------------------------------------
+
+issue_1352 =>
+  c := () ->
+    say "hello"
```

Comment:
This example is IMHO too small, in particular, enhanced type inference could determine the type of this field (`Function unit`) without much problem, while 
```
d := a -> a.x
```
cannot.

Reply:
Hmm, yes that is a good point but expected_err would then change and hence we would notice. Isn't this sufficient?

Reply:
Sorry for a bit more nitpicking:

You have changed the tests to do
```
  a := () -> a.x
```
which is not what I meant. With slightly improved type inference, the type of any lamdba without arguments could e inferred, so `() -> ...` is not a good example, while `a -> ...a...` is better since the a new variable `a` is introduced whose type is unknown. So the example should IMHO be something like
```
  d := a -> a.x
```
or
```
  d := a,b -> a*b
```

Reply:
May thought was `a` can not be inferred even with better type inference because it basically returns itself and hence would need to be known already to be inferred.